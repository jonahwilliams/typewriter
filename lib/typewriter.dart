// Copyright (c) 2017, Jonah Williams. All rights reserved. Use of this source code
// is governed by a BSD-style license that can be found in the LICENSE file.

/// Support for doing something awesome.
///
/// More dartdocs go here.
library typewriter;

export 'analysis/analysis.dart';
export 'annotations/annotations.dart';
export 'generators/generator.dart';

// TODO: Export any libraries intended for clients of this package.
