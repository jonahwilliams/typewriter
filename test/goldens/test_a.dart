library typewriter.testing.test_a;

import 'package:typewriter/typewriter.dart';
import 'dart:convert';

@Json()
class Dog {
  List<String> names;
  double age;
  DateTime birthday;
}
