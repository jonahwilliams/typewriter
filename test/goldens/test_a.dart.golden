part of typewriter.testing.test_a;

const jsonCodec = const JsonCodec();

class DogDecoder extends Converter<Object, Dog> {
  const DogDecoder();

  @override
  Dog convert(Object raw) {
    var input = raw as Map<String, dynamic>;
    var output = new Dog();
    output.names = input["names"].map((x) => x).toList();
    output.age = input["age"];
    output.birthday = DateTime.parse(input["birthday"]);
    return output;
  }
}

class DogEncoder extends Converter<Dog, Object> {
  const DogEncoder();

  @override
  Object convert(Dog input) {
    var output = <String, dynamic>{};
    output["names"] = input.names.map((x) => x).toList();
    output["age"] = input.age;
    output["birthday"] = input.birthday.toIso8601String();
    return output;
  }
}

class DogCodec extends Codec<Dog, Object> {
  const DogCodec();

  @override
  Converter<Object, Dog> get decoder => const DogDecoder();

  @override
  Converter<Dog, Object> get encoder => const DogEncoder();
}
